pkg/cmd/pr/merge/merge.go

```diff
@@ -168,224 +168,326 @@ func NewCmdMerge(f *cmdutil.Factory, runF func(*MergeOptions) error) *cobra.Comm
 		}
+	}
 
-		err = mergePullRequest(httpClient, payload)
-		if err != nil {
-			return err
-		}
+	return nil
+}
 
-		if isTerminal {
-			if payload.auto {
-				method := ""
-				switch payload.method {
-				case PullRequestMergeMethodRebase:
-					method = " via rebase"
-				case PullRequestMergeMethodSquash:
-					method = " via squash"
-				}
-				fmt.Fprintf(opts.IO.ErrOut, "%s Pull request #%d will be automatically merged%s when all requirements are met\n", cs.SuccessIconWithColor(cs.Green), pr.Number, method)
-			} else {
-				action := "Merged"
-				switch payload.method {
-				case PullRequestMergeMethodRebase:
-					action = "Rebased and merged"
-				case PullRequestMergeMethodSquash:
-					action = "Squashed and merged"
-				}
-				fmt.Fprintf(opts.IO.ErrOut, "%s %s pull request #%d (%s)\n", cs.SuccessIconWithColor(cs.Magenta), action, pr.Number, pr.Title)
+// Delete local branch if requested and if allowed.
+func (m *MergeContext) deleteLocalBranch() error {
+	if m.crossRepoPR || m.autoMerge {
+		return nil
+	}
+
+	if m.merged {
+		// prompt for delete
+		if m.opts.InteractiveMode && !m.opts.IsDeleteBranchIndicated {
+			err := prompt.SurveyAskOne(&survey.Confirm{
+				Message: fmt.Sprintf("Pull request #%d was already merged. Delete the branch locally?", m.pr.Number),
+				Default: false,
+			}, &m.deleteBranch)
+			if err != nil {
+				return fmt.Errorf("could not prompt: %w", err)
 			}
+		} else {
+			m.log(fmt.Sprintf("%s Pull request #%d was already merged\n", m.cs.WarningIcon(), m.pr.Number))
 		}
-	} else if !opts.IsDeleteBranchIndicated && opts.InteractiveMode && !crossRepoPR && !opts.AutoMergeEnable {
-		err := prompt.SurveyAskOne(&survey.Confirm{
-			Message: fmt.Sprintf("Pull request #%d was already merged. Delete the branch locally?", pr.Number),
-			Default: false,
-		}, &deleteBranch)
-		if err != nil {
-			return fmt.Errorf("could not prompt: %w", err)
-		}
-	} else {
-		fmt.Fprintf(opts.IO.ErrOut, "%s Pull request #%d was already merged\n", cs.WarningIcon(), pr.Number)
 	}
 
-	if !deleteBranch || crossRepoPR || autoMerge {
+	// the user does not want to delete the branch
+	if !m.deleteBranch {
 		return nil
 	}
 
-	branchSwitchString := ""
-
-	if opts.CanDeleteLocalBranch && localBranchExists {
-		currentBranch, err := opts.Branch()
+	if m.opts.CanDeleteLocalBranch && m.localBranchExists {
+		currentBranch, err := m.opts.Branch()
 		if err != nil {
 			return err
 		}
 
-		var branchToSwitchTo string
-		if currentBranch == pr.HeadRefName {
-			branchToSwitchTo = pr.BaseRefName
-			if branchToSwitchTo == "" {
-				branchToSwitchTo, err = api.RepoDefaultBranch(apiClient, baseRepo)
+		// branch the command was run on is the same as the pull request branch
+		if currentBranch == m.pr.HeadRefName {
+			// if the target branch of the PR is not known, set the current branch to the
+			// default branch of the repository
+			m.currentBranch = m.pr.BaseRefName
+			if m.currentBranch == "" {
```

Comment:
I'm not really sure what the case is for this condition ([prior to refactor](https://github.com/cli/cli/blob/76782744648658e1eee9ee8a666f01d554ac9a16/pkg/cmd/pr/merge/merge.go#L336))
I believe the base of the PR would always be known